getPrimes.js

```diff
@@ -21,6 +21,38 @@ Hint: If the input number is prime you should get the number as the solution. Yo
 care of it.
 */
 
-function getPrimes() {
-	// your code
-}
\ No newline at end of file
+//create an empty array to push the numbers n is divisible by
+//divide n by 2 until it can't anymore
+//divide by the next number (3) if n is still left but cant be didvisible by 2 anymore
+
+//create an empty array to push the numbers n is divisible by
+//divide n by 2 until it can't anymore.
+//divide by the next number (3) if n is still left but cant be didvisible by 2 anymore.
+//push primes to array.
+
+function getPrimes(N) {
+  function isPrime(num) {
+    if (num < 2){
+      return false;
+    }
+	for ( i = 2; i < num; i++){//Check all number less than input num
+		if (num % i === 0){
+			return false;
+        }
+    }
+    return true;
+  }
+
+  var primes=[];
+	for(var k = 2; k < N; k++){
+		if(N % k === 0){
+		   if(isPrime(k) === true){
+             primes.push(k);
+        }
+       }  
+     }
+  console.log(primes);
+}
```

Comment:
This will give a correct solution but not efficient. For each number k that you are checking, you are also checking if it is a prime number. and to check that you are doing k checks in the isPrime function. so for each input number N, you checking if N is divisible by k ~ N times. So each function getPrimes run results in ~ N^2 operation.

A better way to do this would be check if N is divisible by k and if it is divisible, divide and update N until it is not divisible by k, and then proceed to k+1. This way you don't have to check if k is a prime number. A pseudo code approach would look like this:

for i in [2, 3,4, ..... sqrt(N)+1] // check up to square root to reduce redundant checking
        if N % i == 0:
            primes.push(i)
        while N % i == 0:
            N /= i

    return primes